### PR TITLE
📖 [Docs] Merge tilt info into developer.md

### PR DESCRIPTION
**Pull request by bentito**
tilt.md is currently off by itself documenting the developer usage of tilt. It is more desirable to have this just be part of the `developers.md` file and keep all dev docs in one place.

# Description

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

**Review by perdasilva — APPROVED** [review]
Nice one! Thank you!

File: docs/drafts/developer.md [hunk 1]
```diff
@@ -97,4 +93,103 @@ make manifests
 
 **NOTE:** Run `make help` for more information on all potential `make` targets.
 
-More information can be found via the [Kubebuilder Documentation](https://book.kubebuilder.io/introduction.html).
```
Comment (LalatenduMohanty) [inline docs/drafts/developer.md:100]: Not sure if we should remove this link. May be keep it?

Comment (bentito) [reply docs/drafts/developer.md:100]: I think it's sort of out of the blue as-is? Any ideas how to preface it to make it fit somehow?

Comment (bentito) [reply docs/drafts/developer.md:100]: @joelanford advice was to ditch it and I think that makes sense, it's a tie-in to SDK's upstream, right? Seems out of place in a developer doc for operator-controller. wdyt?

**Pull request merged**